File: scheduler.py
#!/usr/bin/python3

from datetime import datetime
import pytz
import schedule
import time
import subprocess

tz = pytz.timezone("Asia/Singapore")

def job():
    print("", flush=True)
    print(f"[{datetime.now(tz).strftime('%d/%m/%Y, %-I:%M:%S %P')}] Running bot.js...", flush=True)
    subprocess.run(["node", "bot.js"])
    print(f"[{datetime.now(tz).strftime('%d/%m/%Y, %-I:%M:%S %P')}] Returned from bot.js!", flush=True)

schedule.every(17).minutes.do(job)

print(f"[{datetime.now(tz).strftime('%d/%m/%Y, %-I:%M:%S %P')}] Starting scheduler...", flush=True)

while 1:
    schedule.run_pending()
    time.sleep(1)

File: bot/scheduler.js
const { ToadScheduler, SimpleIntervalJob, Task } = require('toad-scheduler')
const run = require('./bot.js');

async function execute() {
  console.log(`[${(new Date()).toLocaleString('en-SG', { timeZone: 'Asia/Singapore' })}] Starting bot.js....`);
  await run();
  console.log(`[${(new Date()).toLocaleString('en-SG', { timeZone: 'Asia/Singapore' })}] Execution complete!`);
  console.log();
}

const scheduler = new ToadScheduler()
const task = new Task('Bot', execute);
const job = new SimpleIntervalJob({ minutes: 17, runImmediately: true }, task)
scheduler.addSimpleIntervalJob(job)